fl4health/clients/basic_client.py

```diff
@@ -635,30 +646,64 @@ def train_by_steps(
 
         return loss_dict, metrics
 
-    def validate(self) -> Tuple[float, Dict[str, Scalar]]:
+    def _validate_or_test(
+        self,
+        loader: DataLoader,
+        loss_meter: LossMeter,
+        metric_manager: MetricManager,
+        logging_mode: LoggingMode = LoggingMode.VALIDATION,
+    ) -> Tuple[float, Dict[str, Scalar]]:
         """
-        Validate the current model on the entire validation dataset.
+        Evaluate the model on the given validation or test dataset.
+
+        Args:
+            loader (DataLoader): The data loader for the dataset (validation or test).
+            loss_meter (LossMeter): The meter to track the losses.
+            metric_manager (MetricManager): The manager to track the metrics.
+            logging_mode (LoggingMode): The LoggingMode for whether this evaluation is for validation or test.
+              Default is for validation.
 
         Returns:
-            Tuple[float, Dict[str, Scalar]]: The validation loss and a dictionary of metrics from validation.
+            Tuple[float, Dict[str, Scalar]]: The loss and a dictionary of metrics from evaluation.
         """
+        assert logging_mode in [LoggingMode.VALIDATION, LoggingMode.TEST], "logging_mode must be VALIDATION or TEST"
         self.model.eval()
-        self.val_metric_manager.clear()
-        self.val_loss_meter.clear()
+        metric_manager.clear()
+        loss_meter.clear()
         with torch.no_grad():
-            for input, target in self.val_loader:
+            for input, target in loader:
                 input, target = self._move_input_data_to_device(input), target.to(self.device)
                 losses, preds = self.val_step(input, target)
-                self.val_loss_meter.update(losses)
-                self.val_metric_manager.update(preds, target)
+                loss_meter.update(losses)
+                metric_manager.update(preds, target)
 
         # Compute losses and metrics over validation set
-        loss_dict = self.val_loss_meter.compute().as_dict()
-        metrics = self.val_metric_manager.compute()
-        self._handle_logging(loss_dict, metrics, is_validation=True)
+        loss_dict = loss_meter.compute().as_dict()
+        metrics = metric_manager.compute()
+        self._handle_logging(loss_dict, metrics, logging_mode=logging_mode)
 
         return loss_dict["checkpoint"], metrics
 
+    def validate(self) -> Tuple[float, Dict[str, Scalar]]:
+        """
+        Validate the current model on the entire validation
+            or potentially an entire test dataset if it has been defined.
```

Comment:
The or here sort of implies that we only ever do one or the other. I would suggest changing the "or" to an "and"

Reply:
done!